core/tests/performance_tests/base.py

```diff
@@ -119,3 +143,79 @@ def _test_page_load_time_for_cached_session(self):
         self.assertLessEqual(
             self.page_metrics.get_average_page_load_time_millisecs(),
             self.load_time_limit_cached_ms)
+
+    def _get_driver(self):
+        return self.data_fetcher.setup_driver()
+
+    def _check_login(self, driver):
+        driver.get(self.BASE_URL)
+        time.sleep(5)
+        resulting_url = driver.current_url
+
+        if resulting_url == self.BASE_URL + self.SPLASH_URL_SUFFIX:
+            return False
+
+        return True
+
+    def _login_user(self, driver):
+        driver.get(
+            self.BASE_URL + self.LOGIN_URL_SUFFIX + self.CONTINUE_URL_SUFFIX)
+        elem = driver.find_element_by_name("email")
+        elem.clear()
+        elem.send_keys(self.EMAIL)
+        driver.find_element_by_name('admin').click()
+        driver.find_element_by_id('submit-login').click()
+        time.sleep(5)
```

Comment:
Will this possibly cause flakiness? Will it run consistently across different machines?